erpnext/healthcare/doctype/patient/patient.py

```diff
@@ -93,23 +93,27 @@ def set_missing_customer_details(self):
 			self.language = frappe.db.get_single_value('System Settings', 'language')
 
 	def create_website_user(self):
-		if self.email and not frappe.db.exists('User', self.email):
-			user = frappe.get_doc({
-				'doctype': 'User',
-				'first_name': self.first_name,
-				'last_name': self.last_name,
-				'email': self.email,
-				'user_type': 'Website User',
-				'gender': self.sex,
-				'phone': self.phone,
-				'mobile_no': self.mobile,
-				'birth_date': self.dob
-			})
-			user.flags.ignore_permissions = True
-			user.enabled = True
-			user.send_welcome_email = True
-			user.add_roles('Patient')
-			frappe.db.set_value(self.doctype, self.name, 'user_id', user.name)
+		users = frappe.db.get_all('User', fields=['email', 'mobile_no'], or_filters={'email': self.email, 'mobile_no': self.mobile})
+		if users and users[0]:
+			frappe.throw(_("User exists with Email {}, Mobile {}<br>Please check email / mobile or disable 'Invite as User' to skip creating User")
+				.format(frappe.bold(users[0].email), frappe.bold(users[0].mobile_no)), frappe.DuplicateEntryError)
+
+		user = frappe.get_doc({
+			'doctype': 'User',
+			'first_name': self.first_name,
+			'last_name': self.last_name,
+			'email': self.email,
+			'user_type': 'Website User',
+			'gender': self.sex,
+			'phone': self.phone,
+			'mobile_no': self.mobile,
+			'birth_date': self.dob
+		})
+		user.flags.ignore_permissions = True
+		user.enabled = True
+		user.send_welcome_email = True
+		user.add_roles('Patient')
```

Comment:
A patient is a customer as well when you select link patient to customer.

Reply:
As it's for the portal access, I feel it's best not to add Customer Role considering that not all privileged for Customer will be relevant to Patient

Reply:
OK. But when you have e-commerce enabled for patients like online healthcare packages this is required. Maybe a whitelist will help people like me to override this function?

Reply:
IMO, it would be easier to configure the Patient role without having to override. If it's about having additional permissions for the Patient User. Would that be ok?

Reply:
We set the default role at sign-up as Patient. But the user registered will not be able to place the order online for not being a customer is what I feel. We wanted to override for adding a customer role to a patient registered online.

Reply:
yes, that would help as this is merely a fix to the reported issue, thanks